roles/openshift_openstack/library/os_netpolicy_sg_deletion.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 Red Hat, Inc. and/or its affiliates
+# and other contributors as indicated by the @author tags.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# pylint: disable=unused-wildcard-import,wildcard-import,unused-import,redefined-builtin
+
+''' os_netpolicy_sg_deletions '''
+import keystoneauth1
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_netpolicy_sg_deletions
+short_description: Delete security group associated to the network policy
+description:
+    - Delete the security group associated to the network policy
+author:
+    - "Luis Tomas Bolivar <ltomasbo@redhat.com>"
+'''
+
+RETURN = '''
+'''
+
+
+def main():
+    ''' Main module function '''
+    module = AnsibleModule(
+        argument_spec=dict(
+            sg_id=dict(default=False, type='str'),
```

Comment:
Should this also be defaulted to 'None', the same as in os_namespace_resources_deletion.py?

Reply:
At os_namespace_resources_deletion, the sg_id is optional as you can enable namespace subnet driver without enabling its isolation, which means that you will still create a different subnet per namespace, but without any special isolation (all pods can talk to all pods regardless of their namespace). Therefore, in that case no extra security groups are created for the namespace, and thus the deletion function needs to take that into account to still delete the other namespaces resources. By contrast, at os_netpolicy_sg_deletion, there is no point of accepting empty sg_ids, in that case (as the module just removes the sgs) it is better to simply not call it